From cab7bde238b6a5d3e992e87313c446c86b312e14 Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Fri, 19 Feb 2016 11:44:19 +0000
Subject: [PATCH 18/26] xsplice/test: Convert build-id.o to be relocatable

build-id.o is executable because it is copied from the xen binary which
is an executable. This causes a link failure. Fix by making build-id.o
relocatable.
---
 config/Paths.mk            | 16 ++++++++--------
 config/Stubdom.mk          | 16 +++++-----------
 config/Tools.mk            | 23 ++++++++++-------------
 config/Toplevel.mk         |  2 +-
 tools/config.h             |  8 ++++----
 xen/arch/x86/test/Makefile |  1 +
 6 files changed, 29 insertions(+), 37 deletions(-)

diff --git a/config/Paths.mk b/config/Paths.mk
index 60e0211..8291f5c 100644
--- a/config/Paths.mk
+++ b/config/Paths.mk
@@ -13,11 +13,11 @@
 # http://wiki.xen.org/wiki/Category:Host_Configuration#System_wide_xen_configuration
 
 PACKAGE_TARNAME          := xen
-prefix                   := /usr
-bindir                   := /usr/bin
-sbindir                  := /usr/sbin
-libdir                   := /usr/lib64
-libexecdir               := /usr/libexec
+prefix                   := /usr/local
+bindir                   := /usr/local/bin
+sbindir                  := /usr/local/sbin
+libdir                   := /usr/local/lib
+libexecdir               := /usr/local/lib
 datarootdir              := ${prefix}/share
 mandir                   := ${datarootdir}/man
 docdir                   := ${datarootdir}/doc/${PACKAGE_TARNAME}
@@ -30,11 +30,11 @@ localstatedir            := /var
 sysconfdir               := /etc
 
 LIBEXEC                  := $(libexecdir)/$(PACKAGE_TARNAME)
-LIBEXEC_BIN              := /usr/libexec/xen/bin
+LIBEXEC_BIN              := /usr/local/lib/xen/bin
 LIBEXEC_LIB              := $(LIBEXEC)/lib
 LIBEXEC_INC              := $(LIBEXEC)/include
 
-SHAREDIR                 := /usr/share
+SHAREDIR                 := /usr/local/share
 MAN1DIR                  := $(mandir)/man1
 MAN8DIR                  := $(mandir)/man8
 
@@ -50,7 +50,7 @@ XEN_LOCK_DIR             := /var/lock
 XEN_PAGING_DIR           := /var/lib/xen/xenpaging
 XEN_DUMP_DIR             := /var/lib/xen/dump
 
-XENFIRMWAREDIR           := /usr/libexec/xen/boot
+XENFIRMWAREDIR           := /usr/local/lib/xen/boot
 
 XEN_CONFIG_DIR           := /etc/xen
 XEN_SCRIPT_DIR           := /etc/xen/scripts
diff --git a/config/Stubdom.mk b/config/Stubdom.mk
index a8bb1d3..68ba320 100644
--- a/config/Stubdom.mk
+++ b/config/Stubdom.mk
@@ -1,21 +1,15 @@
-# Prefix and install folder
-prefix              := /usr/local
-PREFIX              := $(prefix)
-exec_prefix         := ${prefix}
-libdir              := ${exec_prefix}/lib
-LIBDIR              := $(libdir)
+-include $(XEN_ROOT)/config/Paths.mk
 
 # Path Programs
-CMAKE               := /usr/bin/cmake
+CMAKE               := /cmake-disabled-in-configure-script
 FETCHER             := /usr/bin/wget -c -O
 
 # A debug build of stubdom? //FIXME: Someone make this do something
 debug               := y
-vtpm = y
 
-STUBDOM_TARGETS     :=  ioemu grub xenstore vtpm vtpmmgr
-STUBDOM_BUILD       :=  ioemu-stubdom pv-grub xenstore-stubdom vtpm-stubdom vtpmmgr-stubdom
-STUBDOM_INSTALL     :=  install-ioemu install-grub install-xenstore install-vtpm install-vtpmmgr
+STUBDOM_TARGETS     :=  grub xenstore ioemu
+STUBDOM_BUILD       :=  pv-grub xenstore-stubdom ioemu-stubdom
+STUBDOM_INSTALL     :=  install-grub install-xenstore install-ioemu
 
 ZLIB_VERSION        := 1.2.3
 ZLIB_URL            := $(XEN_EXTFILES_URL)
diff --git a/config/Tools.mk b/config/Tools.mk
index 76a6066..e485ba1 100644
--- a/config/Tools.mk
+++ b/config/Tools.mk
@@ -5,9 +5,6 @@ ifeq ($(CONFIG_RUMP),y)
 XEN_OS              := NetBSDRump
 endif
 
-# A debug build of tools?
-debug               := @debug@
-
 # Tools path
 BISON               := /usr/bin/bison
 FLEX                := /usr/bin/flex
@@ -52,30 +49,30 @@ GIT_HTTP            := n
 
 # Optional components
 XENSTAT_XENTOP      := y
-OCAML_TOOLS         := y
-FLASK_POLICY        := n
+OCAML_TOOLS         := n
+FLASK_POLICY        := y
 CONFIG_OVMF         := n
 CONFIG_ROMBIOS      := y
-CONFIG_SEABIOS      := n
+CONFIG_SEABIOS      := y
 CONFIG_QEMU_TRAD    := y
-CONFIG_QEMU_XEN     := n
-CONFIG_BLKTAP2      := n
+CONFIG_QEMU_XEN     := y
+CONFIG_BLKTAP2      := y
 CONFIG_QEMUU_EXTRA_ARGS:= 
 CONFIG_REMUS_NETBUF := n
 
 CONFIG_SYSTEMD      := y
-SYSTEMD_CFLAGS      :=  
-SYSTEMD_LIBS        := -lsystemd-daemon  
+SYSTEMD_CFLAGS      := 
+SYSTEMD_LIBS        := -lsystemd 
 XEN_SYSTEMD_DIR     := $(prefix)/lib/systemd/system/
 XEN_SYSTEMD_MODULES_LOAD := $(prefix)/lib/modules-load.d/
 
 LINUX_BACKEND_MODULES := xen-evtchn xen-gntdev xen-gntalloc xen-blkback xen-netback xen-pciback evtchn gntdev netbk blkbk xen-scsibk usbbk pciback xen-acpi-processor blktap2
 
 #System options
-ZLIB                :=  -DHAVE_BZLIB -lbz2 -DHAVE_LZMA -llzma
+ZLIB                :=  -DHAVE_BZLIB -lbz2
 CONFIG_LIBICONV     := n
-CONFIG_GCRYPT       := y
-EXTFS_LIBS          := -lext2fs
+CONFIG_GCRYPT       := n
+EXTFS_LIBS          := 
 CURSES_LIBS         := -lncurses
 TINFO_LIBS          := -ltinfo
 ARGP_LDFLAGS        := 
diff --git a/config/Toplevel.mk b/config/Toplevel.mk
index ced7d97..7f8b842 100644
--- a/config/Toplevel.mk
+++ b/config/Toplevel.mk
@@ -1 +1 @@
-SUBSYSTEMS               :=  xen tools docs
+SUBSYSTEMS               :=  xen tools stubdom docs
diff --git a/tools/config.h b/tools/config.h
index fb7b897..363ff9c 100644
--- a/tools/config.h
+++ b/tools/config.h
@@ -5,7 +5,7 @@
 /* #undef ENABLE_PARTIAL_DEVICE_TREE */
 
 /* Blktap2 enabled */
-/* #undef HAVE_BLKTAP2 */
+#define HAVE_BLKTAP2 1
 
 /* Define to 1 if you have the declaration of `fdt_first_subnode', and to 0 if
    you don't. */
@@ -29,7 +29,7 @@
 #define HAVE_INTTYPES_H 1
 
 /* Define to 1 if you have the `aio' library (-laio). */
-/* #undef HAVE_LIBAIO */
+#define HAVE_LIBAIO 1
 
 /* Define to 1 if you have the `crypto' library (-lcrypto). */
 #define HAVE_LIBCRYPTO 1
@@ -92,7 +92,7 @@
 #define INCLUDE_CURSES_H <ncurses.h>
 
 /* Define extfs header to use */
-#define INCLUDE_EXTFS_H <ext2fs/ext2fs.h>
+/* #undef INCLUDE_EXTFS_H */
 
 /* libutil header file name */
 /* #undef INCLUDE_LIBUTIL_H */
@@ -116,7 +116,7 @@
 #define PACKAGE_VERSION "4.6"
 
 /* Qemu Xen path */
-#define QEMU_XEN_PATH "/usr/lib64/xen/bin/qemu-system-i386"
+#define QEMU_XEN_PATH "/usr/local/lib/xen/bin/qemu-system-i386"
 
 /* Define to 1 if you have the ANSI C header files. */
 #define STDC_HEADERS 1
diff --git a/xen/arch/x86/test/Makefile b/xen/arch/x86/test/Makefile
index e80c68d..f2eef14 100644
--- a/xen/arch/x86/test/Makefile
+++ b/xen/arch/x86/test/Makefile
@@ -46,6 +46,7 @@ build_id.o:
 	# Need to clear the CODE when the build_id.o is put in the .data
 	$(OBJCOPY) --set-section-flags=.note=alloc,load,data  \
 		   --rename-section=.note=.xsplice.depends .$@.0 $@
+	elfedit --output-type=rel $@
 	rm -f .$@.0
 
 .PHONY: xsplice
-- 
2.4.3

